src/clusterfuzz/_internal/system/archive.py

```diff
@@ -197,137 +339,28 @@ def unpack(archive_path,
   if not os.path.exists(archive_path):
     logs.log_error('Archive %s not found.' % archive_path)
     return False
+  reader = get_archive_reader(archive_path=archive_path)
+  assert reader is not None
```

Comment:
It seems that every call to `get_archive_reader()` assumes this is true. Let's extract this check into `get_archive_reader()` and add some information to the exception (filename)?